zstacklib/zstacklib/utils/upload_task.py

```
@@ -0,0 +1,350 @@
+        if up.slice_offset >= up.total_size:
+            raise Exception('invalid slice offset header: %s, total_size: %d' %
+                            (up.slice_offset, up.total_size))
+
+        return up
+
+    def update_task_by_upload_param(self, param, task):
+        if task.lastError:
+            self._fail_task(task, task.lastError)
+
+        if task.completed:
+            raise Exception('image[uuid: %s] upload has completed' % param.image_uuid)
```

Ruff (0.13.1)

Create your own exception

(TRY002)